src/strategies/alchemix/strategy-alcx-farm-base.sol

```diff
@@ -0,0 +1,132 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.6.7;
+
+import "../strategy-base.sol";
+import "../../interfaces/alcx-farm.sol";
+
+abstract contract StrategyAlcxFarmBase is StrategyBase {
+    // Token addresses
+    address public constant alcx = 0xdBdb4d16EdA451D0503b854CF79D55697F90c8DF;
+    address public constant stakingPool = 0xAB8e74017a8Cc7c15FFcCd726603790d26d7DeCa;
+
+    // How much ALCX tokens to keep?
+    uint256 public keepAlcx = 0;
+    uint256 public constant keepAlcxMax = 10000;
+
+    uint256 public poolId;
+
+    constructor(
+        uint256 _poolId,
+        address _lp,
+        address _governance,
+        address _strategist,
+        address _controller,
+        address _timelock
+    )
+        public
+        StrategyBase(
+            _lp,
+            _governance,
+            _strategist,
+            _controller,
+            _timelock
+        )
+    {
+        poolId = _poolId;
+    }
+    
+
+    function balanceOfPool() public view override returns (uint256) {
+        uint256 amount = IStakingPools(stakingPool).getStakeTotalDeposited(address(this), poolId);
+        return amount;
+    }
+
+    function getHarvestable() public view returns (uint256) {
+        return IStakingPools(stakingPool).getStakeTotalUnclaimed(address(this), poolId);
+    }
+
+    // **** Setters ****
+
+    function deposit() public override {
```

Comment:
Not sure  if pickle has moved away from this, but the public earn interaction is something we moved away in v1 before sunsetting some of the vaults, we realized there are several problems that arise when user can force the strategy to invest, this may not be the case here depending of the state of stakingPool, but is smth that pickle may want to explore
See:
https://github.com/yearn/yearn-protocol/blob/develop/contracts/strategies/StrategyDAI3poolV2.sol

The other option is to pull funds during harvest instead of public earn investing all the way to staking pool

Reply:
well actually, calling earn function is not that important because it's called every time you deposit. So I think I can make `earn` function an internal function.

Reply:
Sorry.... I neglected to hit "submit review" and the above message has been stuck in limbo.